packages/core-ethereum/src/ethereum.ts

```diff
@@ -735,6 +754,43 @@ const provider = networkInfo.provider.startsWith('http')
     return new Promise(() => "");
   }
 
+  /**
+   * Initiates a transaction that registers a safe address
+   * @param safeAddress address of safe
+   * @param txHandler handler to call once the transaction has been published
+   * @returns a Promise that resolves with the transaction hash
+   */
+  const registerSafeByNode = async (
+    safeAddress: Address,
+    txHandler: (tx: string) => DeferType<string>
+  ): Promise<Receipt> => {
+    log('Register a node to safe %s globally', safeAddress.to_hex())
+    let sendResult: SendTransactionReturn
+    let error: unknown
+
+    try {
+      const registerSafeByNodeEssentialTxPayload = buildEssentialTxPayload(
+        0,
+        nodeSafeRegistry,
+        'registerSafeByNode',
+        safeAddress.to_hex()
+      )
+      sendResult = await sendTransaction(checkDuplicate, registerSafeByNodeEssentialTxPayload, txHandler)
+    } catch (err) {
+      error = err
+    }
+
+    switch (sendResult.code) {
+      case SendTransactionStatus.SUCCESS:
+        return sendResult.tx.hash
+      case SendTransactionStatus.DUPLICATE:
+        throw new Error(`Failed in sending registerSafeByNode transaction because transaction is a duplicate`)
+      default:
+        throw new Error(`Failed in sending registerSafeByNode transaction due to ${error}`)
+    }
+    // TODO: catch race-condition
```

Comment:
Is this one resolved?

Reply:
This race-condition is not resolved because it's a slim chance hitting it. 
This "catch race-condition" is for the case where a hopr node restarts at the point when the "register node safe" transaction is not confirmed with finality. This restart triggers the same transaction, which may run into concurrency issue.